manufacturing-client/src/main.rs

```diff
@@ -937,3 +979,52 @@ impl KeyReference {
         }
     }
 }
+
+const IPV6_DEFAULT: &str = "00000000000000000000000000000000"; //DevSkim: ignore DS173237
+const IPV4_DEFAULT: &str = "00000000";
+
+fn get_default_network_iface() -> Result<Option<String>, std::io::Error> {
+    // Check IPv4 addresses from /proc/net/route
+    if let Ok(file) = std::fs::File::open("/proc/net/route") {
+        let reader = BufReader::new(file);
+
+        for line in reader.lines().skip(1) {
+            let line = line?;
+            let fields: Vec<_> = line.split_whitespace().collect();
+            if fields.is_empty() {
+                continue;
+            }
+            if fields[1] == IPV4_DEFAULT && fields[0] != "lo" {
+                let iface = fields[0].to_string();
+                log::info!("Default network interface is ipv4 based {}", iface);
+                return Ok(Some(iface));
+            }
+        }
+    }
+
+    // Check IPv6 addresses from /proc/net/ipv6_route
+    // FYI: For IPv4 addresses, DNS maintains so-called “A” records, for “Address”.
+    // The IPv6 equivalent is the “AAAA” record, for “Address four times longer”.
+    // so you can see fields being compared to a string which is fourtimes longer
+    // for ipv4 than for ipv6.
+    // For ipv6_route file and columns meaning ,refer https://tldp.org/HOWTO/Linux+IPv6-HOWTO/ch11s04.html
+
+    if let Ok(file) = std::fs::File::open("/proc/net/ipv6_route") {
+        let reader = BufReader::new(file);
+
+        for line in reader.lines().skip(1) {
+            let line = line?;
+            let fields: Vec<_> = line.split_whitespace().collect();
+            if fields.is_empty() {
+                continue;
+            }
+            if fields[1] == IPV6_DEFAULT && fields[9] != "lo" {
+                let iface = fields[0].to_string();
+                log::info!("Default network interface is ipv6 based {}", iface);
+                return Ok(Some(iface));
+            }
```

Comment:
This logic is not going to work for parsing `/proc/net/ipv6_route`, because the layout of the fields is quite different from `/proc/net/route`

Skipping the first line of the output is likely going to drop one of the IPv6 loopback addresses; on two of my Fedora systems the line looks like:

`00000000000000000000000000000001 80 00000000000000000000000000000000 00 00000000000000000000000000000000 00000100 00000001 00000000 00000001       lo`

`fields[0]` in this table is going to be "IPv6 destination network displayed in 32 hexadecimal chars without colons as separator", so it is not the right field to use for the interface name.

It's also possible for an interface to have a single IPv6 address that is "link local" and will never be routed, so it is unlikely to be usable for broader communication. I would think we would need to check for this kind of address before we decide to use the interface name.

Reply:
Agree!
I think instead of extracting the network interface at low level,   it will be better to use some networking related crate/s which provides info on the active ipv6 network interface etc . I would like to convert this PR to draft, go back and check for a proper crate so that even if something changes with the kernel files (e.g. columns or way of representation this will fail!) whereas with crate it will be(probably) handled in case of any changes .
CC: @puiterwijk @7flying

Reply:
my only concern is that the crate (if any) should ideally be already packaged in Fedora, as well as its dependencies

Reply:
is there a standard network crate that provides all this information for both IPv4 and IPv6 that is widely used?

Reply:
last time when I had looked for such crates, there were many options but
-  `ipnet` seems popular and has stable releases (https://lib.rs/crates/ipnet)
- also std::net in combination with ipnet can be used